### PR TITLE
samples: matter: fix initialization order

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Make sure that all the initialization code happens prior to
the point in which the CHIP thread is started to guarantee
no data races can occur in that phase.

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

### CODE REVIEW

File: samples/matter/light_bulb/src/app_task.cpp [hunk 1]
```diff
@@ -77,8 +80,35 @@ chip::OTARequestor sOTARequestor;
...
+		return err;
+	}
+
+	err = ThreadStackMgr().InitThreadStack();
+	if (err != CHIP_NO_ERROR) {
+		LOG_ERR("ThreadStackMgr().InitThreadStack() failed");
+		return err;
+	}
+
+	err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
```
Comment (kkasperczyk-no) [inline samples/matter/light_bulb/src/app_task.cpp:106]: Maybe we could change device type to FTD by the way of this change?

**Review by kkasperczyk-no — APPROVED** [review]
LGTM. 
Side comment is that maybe we should not assume `kThreadDeviceType_MinimalEndDevice` in the samples, but set device type depending on the `CONFIG_OPENTHREAD_FTD` or `CONFIG_OPENTHREAD_MTD` configs. That's just an observation and we can create separate task for that.

File: samples/matter/light_bulb/src/app_task.cpp (hunk 1 above)
Comment (Damian-Nordic) [reply samples/matter/light_bulb/src/app_task.cpp:106]: Let's do it a separate PR if there's a plan to do it. It's not really related to the initialization order after all.

**Comment by markaj-nordic** [comment]
I am just wondering, maybe we should think about some mechanism for forcing the proper init order for all our samples by generic `Init()` or template method pattern. That could save us from similar issues in the future as well.

**Review by markaj-nordic — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
@markaj-nordic Matter community doesn't like the idea of a common initialization and they prefer making all the apps flexible and configure everything as they want. In fact, some examples in the upstream do the initialization like us, while the others run it on the CHIP thread, while controller applications typically run the CHIP event loop on the main thread :)

We can discuss, however, if we want to prepare such a helper just for our samples. Perhaps, we could have a helper for initializing the Matter stack and control its behaviour with Kconfig, and another helper for starting the CHIP event loop.

**Pull request merged**